connect/api/src/main/java/org/apache/kafka/connect/connector/ConnectRecord.java

```diff
@@ -34,18 +39,33 @@
     private final Schema valueSchema;
     private final Object value;
     private final Long timestamp;
+    private final Headers headers;
 
     public ConnectRecord(String topic, Integer kafkaPartition,
                          Schema keySchema, Object key,
                          Schema valueSchema, Object value,
                          Long timestamp) {
+        this(topic, kafkaPartition, keySchema, key, valueSchema, value, timestamp, null);
+    }
+
+    public ConnectRecord(String topic, Integer kafkaPartition,
+                         Schema keySchema, Object key,
+                         Schema valueSchema, Object value,
+                         Long timestamp, Iterable<Header> headers) {
         this.topic = topic;
         this.kafkaPartition = kafkaPartition;
         this.keySchema = keySchema;
         this.key = key;
         this.valueSchema = valueSchema;
         this.value = value;
         this.timestamp = timestamp;
+        if (headers == null) {
```

Comment:
I guess I was just trying to be safe in case someone is passing in a reference they didn't know is null. Seemed easy enough to handle, especially since not handling such a case means the connector task dies. But I can clean it up a bit since the `ConnectHeaders` constructor can take a null value.